### ISSUE TITLE
Add py3-ldap3 to Docker image

### ISSUE THREAD

**Issue by unmacaque** [tomsquest/docker-radicale#162]
Since Radicale 3.3.0.0, authentication with LDAP is supported - again.

References:
- https://github.com/Kozea/Radicale/releases/tag/v3.3.0
- https://github.com/Kozea/Radicale/wiki/LDAP-authentication

This does not yet work with this image yet, as Radicale will fail to start up, because Python module `ldap3` is missing.

Would it be possible to pre-bundle [py3-ldap3](https://pkgs.alpinelinux.org/package/v3.19/community/x86_64/py3-ldap3) with this image, so LDAP support works out of the box?

**Comment by tomsquest** [comment]
Hi @unmacaque ,

I am unsure about adding a dependency in this image. It can make it brittle and a bit heavier.
The more is added, the more change need to be tracked and... fixed.

Can you consider extending the image using https://github.com/tomsquest/docker-radicale?tab=readme-ov-file#extending-the-image ?

**Comment by alceasan** [comment]
I'd ask to reopen this again, maybe the extended option is no necessary anymore for LDAP auth, but then I'm not sure how to configure it. I used the radicale-extended option including the radicale3-auth-ldap library for LDAP auth, but builiding the extended image after 3.3 update results in app not working anymore:

```
[2024-11-05 07:44:43 +0000] [7] [INFO] Logging of backtrace is disabled in this loglevel
[2024-11-05 07:44:43 +0000] [7] [DEBUG] Logging of backtrace is disabled by option in this loglevel
[2024-11-05 07:44:43 +0000] [7] [INFO] Loaded default config
[2024-11-05 07:44:43 +0000] [7] [INFO] Loaded config file '/config/config'
[2024-11-05 07:44:43 +0000] [7] [INFO] Starting Radicale
[2024-11-05 07:44:43 +0000] [7] [INFO] auth type is 'radicale3_auth_ldap'
[2024-11-05 07:44:43 +0000] [7] [CRITICAL] An exception occurred during server startup: option already exists in 'auth': 'ldap_base'

```

It stills works building with the 3.2 version.

**Comment by tomsquest** [comment]
Hi @alceasan ,

Hard to diagnose without a LDAP system.

Could it be related to the `'ldap_base` in your config?

**Comment by tomsquest** [comment]
@TilBlechschmidt ,

Thanks for your suggestions.

You totally made the point: maintenance, what to test, what about security, release tag, how to document, what if there is another variant with oauth (should it include ldap or not?)...

Still, for now, we are talking about adding 1 or 2 dependencies for LDAP.

Questions:
- Does `py3-ldap3` is needed along installing `ldap3`? (one in the system, one in the venv)?
- How much of increase of the size of the image?

**Comment by TilBlechschmidt** [comment]
1. It is not
2. ~5 MB uncompressed and [~1 MB compressed](https://hub.docker.com/r/themegatb/docker-radicale/tags) so <4%

I added `py3-ldap` initially based on the original post but it seems that the pip package inside the venv suffices.

**Comment by tomsquest** [comment]
Hi  @TilBlechschmidt ,

I created a PR at https://github.com/tomsquest/docker-radicale/pull/169 (just adding ldap3).

Can you build it to validate it's working in your use case?

I will then merge and release it.

Thanks

**Comment by TilBlechschmidt** [comment]
Just tested it, works like a charm 🙂

Thank you @tomsquest !

**Comment by tomsquest** [comment]
Released as 3.3.1.1 :+1: 
https://github.com/tomsquest/docker-radicale/releases/tag/3.3.1.1
